test/Microsoft.AspNetCore.Blazor.Build.Test/BindRazorIntegrationTest.cs

```diff
@@ -0,0 +1,491 @@
+                frame => AssertFrame.Element(frame, "input", 4, 0),
+                frame => AssertFrame.Attribute(frame, "type", "text", 1),
+                frame => AssertFrame.Attribute(frame, "value", "42", 2),
+                frame => AssertFrame.Attribute(frame, "onchange", typeof(UIEventHandler), 3),
+                frame => AssertFrame.Whitespace(frame, 4));
+        }
+
+        [Fact]
+        public void Render_BuiltIn_BindToInputCheckbox_WritesAttributes()
+        {
+            // Arrange
+            var component = CompileToComponent(@"
+@addTagHelper *, TestAssembly
+<input type=""checkbox"" bind=""@Enabled"" />
+@functions {
+    public bool Enabled { get; set; }
+}");
+
+            // Act
+            var frames = GetRenderTree(component);
+
+            // Assert
+            Assert.Collection(
+                frames,
+                frame => AssertFrame.Element(frame, "input", 4, 0),
+                frame => AssertFrame.Attribute(frame, "type", "checkbox", 1),
+                frame => AssertFrame.Attribute(frame, "value", "False", 2),
+                frame => AssertFrame.Attribute(frame, "onchange", typeof(UIEventHandler), 3),
+                frame => AssertFrame.Whitespace(frame, 4));
+        }
+
+        [Fact]
+        public void Render_BindToElementFallback_WritesAttributes()
+        {
+            // Arrange
+            var component = CompileToComponent(@"
+@addTagHelper *, TestAssembly
+<input type=""text"" bind-value-changed=""@ParentValue"" />
+@functions {
+    public int ParentValue { get; set; } = 42;
+}");
+
+            // Act
+            var frames = GetRenderTree(component);
+
+            // Assert
+            Assert.Collection(
+                frames,
+                frame => AssertFrame.Element(frame, "input", 4, 0),
+                frame => AssertFrame.Attribute(frame, "type", "text", 1),
+                frame => AssertFrame.Attribute(frame, "value", "42", 2),
+                frame => AssertFrame.Attribute(frame, "changed", typeof(UIEventHandler), 3),
+                frame => AssertFrame.Whitespace(frame, 4));
+        }
+
+        [Fact]
+        public void Render_BindToElementFallback_WithFormat_WritesAttributes()
+        {
+            // Arrange
+            var component = CompileToComponent(@"
+@addTagHelper *, TestAssembly
+<input type=""text"" bind-value-changed=""@CurrentDate"" format-value=""MM/dd"" />
+@functions {
+    public DateTime CurrentDate { get; set; } = new DateTime(2018, 1, 1);
+}");
+
+            // Act
+            var frames = GetRenderTree(component);
+
+            // Assert
+            Assert.Collection(
+                frames,
+                frame => AssertFrame.Element(frame, "input", 4, 0),
+                frame => AssertFrame.Attribute(frame, "type", "text", 1),
+                frame => AssertFrame.Attribute(frame, "value", "01/01", 2),
+                frame => AssertFrame.Attribute(frame, "changed", typeof(UIEventHandler), 3),
```

Comment:
Oh, I see that is covered in the integration tests below.

Reply:
Yeah, I left the existing integration tests in place so that I could ensure that they all still worked with no modifications other than the source changes. I plan to de-dupe them when we remove `@Bind`